src/main/java/de/tum/in/www1/artemis/service/dto/UserPublicInfoDTO.java

```diff
@@ -0,0 +1,165 @@
+package de.tum.in.www1.artemis.service.dto;
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.User;
+
+/**
+ * A DTO representing a user with the minimal information allowed to be seen by other users in a course
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class UserPublicInfoDTO {
```

Comment:
I think we switched to records?
https://www.baeldung.com/java-record-keyword

Reply:
Records are just syntactic sugar. I see no reasons to change this here.